extension/test/data_test.ts

```diff
@@ -69,4 +69,166 @@ describe('data.ts', function () {
       ).to.include.something.like({ entry: /^ＤＶＤ-ＲＯＭ .*/ });
     });
   });
+
+  describe('makeText', function () {
+    describe('with a word dict entry', function () {
+      // wordDictEntry was generated with: rcxDict.wordSearch('あ', /* doNames= */ false)
+      const wordDictEntry = {
+        kanji: '',
+        onkun: '',
+        nanori: '',
+        bushumei: '',
+        misc: {},
+        eigo: '',
+        hasNames: false,
+        data: [
+          { entry: 'あ /(int) (1) ah/oh/(int) (2) hey!/', reason: undefined },
```

Comment:
Similar to above, can we just exclude reason or does it have to be undefined to get the same answer. (maybe in that case add a note to one of the descriptors)

Reply:
We can't exclude the reason field because the reason field's type is written ["reason: string | undefined"](https://github.com/melink14/rikaikun/blob/main/extension/data.ts#L55), but if we change the type to "reason?: string" then we will be able to exclude the "reason: undefined" part in the tests. Let me know if I can change the type to "reason?: string" or if I should just leave the type alone for now in this pull request.

Reply:
Interesting, I thought excluding a field was the same as the field being undefined in typescript. There are other options (mentioned in the other comment) for avoiding irrelevant properties but this one is short so I would leave as is.

I think in the long term this property being optional is correct. I'm not sure why an explicit `undefined` type would ever by preferred off the top of my head.

Reply:
If you use `{ reason?: string }`, then `{ reason: undefined }` and `{}` are both allowed.
If you use `{ reason: string | undefined }`, then `{ reason: undefined }` is allowed but `{}` is not allowed.

I checked in the TypeScript playground [here](https://www.typescriptlang.org/play?noUnusedParameters=true&ssl=10&ssc=1&pln=1&pc=1#code/C4TwDgpgBA8mwEsD2A7AhgGwEoQMZICcATKAXigG8oCI0BnVAfgC4o7gCEUBzKAXwDcAKCH4U7KEnjJ02PISIBGVnESpMOfMTKVqtBilYBXFEQgAzLhBKDRqCVLWzNCgEwrp6uVpLkKtoVBIKABVUwsrIhdtPz16VFZ2Th4oAB8oEzNLFGt+YTtxYAzw7OtopVYwrMjynSoaeMNi6pybYTEJTIjW8vdQkpr5GMpBKAB6Magg6AgCAkIhIA).

Reply:
The only benefit I could think of for using `{ reason: string | undefined }` would be if you have some code that depends on the property always being there, for example maybe if you have code that checks the number of entries in the object for some reason (maybe something like `Object.keys(obj).length` which is expected to always be the same), but this seems unlikely, so it's probably fine to use `{ reason?: string }`.